cmd/server/handler_client_factory.go

```diff
@@ -27,33 +27,13 @@ import (
 	"github.com/ory/herodot"
 	"github.com/ory/hydra/client"
 	"github.com/ory/hydra/config"
-	"github.com/ory/sqlcon"
 )
 
 func newClientManager(c *config.Config) client.Manager {
 	ctx := c.Context()
 
-	switch con := ctx.Connection.(type) {
-	case *config.MemoryConnection:
-		expectDependency(c.GetLogger(), ctx.Hasher)
-		return client.NewMemoryManager(ctx.Hasher)
-	case *sqlcon.SQLConnection:
-		expectDependency(c.GetLogger(), ctx.Hasher, con.GetDatabase())
-		return &client.SQLManager{
-			DB:     con.GetDatabase(),
-			Hasher: ctx.Hasher,
-		}
-	case *config.PluginConnection:
-		if m, err := con.NewClientManager(); err != nil {
-			c.GetLogger().Fatalf("Could not load client manager plugin %s", err)
-		} else {
-			return m
-		}
-		break
-	default:
-		panic("Unknown connection type.")
-	}
-	return nil
+	expectDependency(c.GetLogger(), ctx.Hasher, ctx.Connection)
```

Comment:
We don't really need the ctx.Hash/logger dependencies here. This could be the case for plugins for example (they might use a different hasher). I think it would make more sense to put `expectDependency` in the concrete implementations.

Reply:
Hmmm - so move the function and all calls to config and call it as required there, e.g. for the SQL Manager? I'm not entirely sure why these were added so I just copied them back as best I could.

Reply:
The problem is that some services have dependencies to others. In some cases (in the past) these had not been properly instantiated during the set up of a particular service/manager. This caused nil point exceptions (see #928 for example). To solve that I added this little helper that makes sure that no dependency is nil.

Adding this to the SQL or Memory Managers (where required) prevents that.